src/routes/auth/auth.test.ts

```diff
@@ -116,20 +113,30 @@ it('should complain about incorrect email', async () => {
   expect(status).toEqual(400)
 })
 
+function expectValidJWT(jwtToken: string) {
+  const decodedJwt = JWT.decode(jwtToken) as Token
+  expect(decodedJwt[JWT_CLAIMS_NAMESPACE]).toBeObject()
+  // Test if the custom claims work
+  expect(decodedJwt[JWT_CLAIMS_NAMESPACE]['x-hasura-name']).toEqual('Test name')
+}
+
 it('should sign the user in', async () => {
   const { body, status } = await request.post('/auth/login').send({ email, password })
-  // Save JWT token to globally scoped varaible.
-  jwtToken = body.jwt_token
   expect(status).toEqual(200)
   expect(body.jwt_token).toBeString()
   expect(body.jwt_expires_in).toBeNumber()
+  expectValidJWT(body.jwt_token)
 })
 
-it('should decode a valid custom user claim', async () => {
-  const decodedJwt = JWT.decode(jwtToken) as Token
-  expect(decodedJwt[JWT_CLAIMS_NAMESPACE]).toBeObject()
-  // Test if the custom claims work
-  expect(decodedJwt[JWT_CLAIMS_NAMESPACE]['x-hasura-name']).toEqual('Test name')
+it('should sign the user in using the admin secret', async () => {
+  const { body, status } = await request.post('/auth/login')
+    .send({ email, password: '' })
+    .set(AdminSecretHeader, ADMIN_SECRET)
```

Comment:
This .set is not working, i don't know why

Reply:
That's strange. The syntax looks OK.

Reply:
Oh, apparently the order matters. Look here: https://github.com/visionmedia/supertest/issues/398#issuecomment-366403045

Reply:
Wait, you got the correct order. :/

Reply:
I tried both without success :/

Reply:
I still have this issue